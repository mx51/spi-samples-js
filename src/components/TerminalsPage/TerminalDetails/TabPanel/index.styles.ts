import { createStyles, makeStyles, Theme } from '@material-ui/core';
import { drawerWidth } from '../../../../definitions/constants/commonConfigs';
import { IFlowPanel } from '../../../FlowPanel/interfaces';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      width: '100%',
      display: 'flex',
      '& .MuiTabs-indicator': {
        backgroundColor: theme.palette.primary.main,
      },
    },
    panel: {
      flexGrow: 1,
      padding: theme.spacing(6),
      transition: theme.transitions.create('margin', {
        easing: theme.transitions.easing.sharp,
        duration: theme.transitions.duration.leavingScreen,
      }),
      marginRight: `${-drawerWidth}%`,
    },
    panelShift: {
      transition: theme.transitions.create('margin', {
        easing: theme.transitions.easing.easeOut,
        duration: theme.transitions.duration.enteringScreen,
      }),
      marginRight: 0,
    },
    tabPanelContainer: {
      display: 'inherit',
      width: '100%',
    },
    [theme.breakpoints.between(960, 1025)]: {
      root: {
        '&>main': {
          padding: theme.spacing(6, 2),
        },
        '& form': {
          marginBottom: (showFlowPanel: IFlowPanel) => (showFlowPanel ? 0 : -theme.spacing(2)),
        },
      },
    },
    [theme.breakpoints.between(600, 959)]: {
      root: {
        '&>main': {
          padding: theme.spacing(6, 0),
        },
      },
      pairStatusContainer: {
        flexBasis: (showFlowPanel: IFlowPanel) => (showFlowPanel ? '100%' : '33.333333%'),
        maxWidth: (showFlowPanel: IFlowPanel) => (showFlowPanel ? '100%' : '33.333333%'),
      },
    },
    [theme.breakpoints.down(600)]: {
      root: {
        '&>main': {
          padding: theme.spacing(6, 0),
        },
        '& form': {
          marginBottom: theme.spacing(2),
        },
      },
      tabPanelContainer: {
        flexBasis: '100%',
        width: '100%',
        maxWidth: '100%',
        '&>div': {
          paddingRight: 0,
        },
      },
    },
    // tab showFlowPanel stylings
    text: {
      fontWeight: 300,
    },
    flowButton: {
      color: 'inherit',
      background: 'transparent',
      border: 'none',
      cursor: 'pointer',
    },
    flowIcon: {
      cursor: 'pointer',
    },
    // receipt panel stylings
    detailsPanelContainer: {
      width: '100%',
      display: 'flex',
    },
    detailsPanel: {
      backgroundColor: theme.palette.common.white,
      border: `1px solid ${theme.palette.secondary.main}`,
      borderRadius: theme.spacing(0.5),
      boxSizing: 'border-box',
      flexGrow: 1,
      marginRight: `${-drawerWidth}%`,
      padding: theme.spacing(3),
      transition: theme.transitions.create('margin', {
        easing: theme.transitions.easing.sharp,
        duration: theme.transitions.duration.leavingScreen,
      }),
    },
    detailsPanelShift: {
      transition: theme.transitions.create('margin', {
        easing: theme.transitions.easing.easeOut,
        duration: theme.transitions.duration.enteringScreen,
      }),
      marginRight: 0,
    },
    receiptPanel: {
      flexShrink: 0,
      marginLeft: theme.spacing(4),
      width: `${drawerWidth}%`,
    },
    receiptPanelOpened: {
      backgroundColor: theme.palette.common.white,
      border: `1px solid ${theme.palette.secondary.main}`,
      borderRadius: theme.spacing(0.5),
      boxSizing: 'border-box',
      flexShrink: 0,
      marginLeft: theme.spacing(4),
      width: `${drawerWidth}%`,
    },
    preContent: {
      whiteSpace: 'pre-wrap',
    },
  })
);

export default useStyles;
