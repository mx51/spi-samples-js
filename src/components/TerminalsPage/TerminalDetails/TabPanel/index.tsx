import React from 'react';
import { Container } from '@material-ui/core';
import Box from '@material-ui/core/Box';
import Button from '@material-ui/core/Button';
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';
import KeyboardArrowLeftIcon from '@material-ui/icons/KeyboardArrowLeft';
import KeyboardArrowRightIcon from '@material-ui/icons/KeyboardArrowRight';
import FlowPanel from '../../../FlowPanel';
import { TabPanelProps } from '../interfaces';
import ReceiptPanel from '../ReceiptPanel';
import useStyles from './index.styles';

export default function TabPanel({
  children,
  flow,
  index,
  setFlow,
  subtitle,
  title,
  value,
  receiptToggle,
  terminal,
}: TabPanelProps): React.ReactElement {
  const classes = useStyles({ flow, receiptToggle });

  return (
    <>
      {value === index && (
        <div className={classes.root}>
          <div className={flow ? `${classes.panel} ${classes.panelShift}` : classes.panel}>
            <Container maxWidth="lg">
              <Grid className={classes.detailsPanelContainer}>
                <div
                  className={
                    receiptToggle ? `${classes.detailsPanel} ${classes.detailsPanelShift}` : classes.detailsPanel
                  }
                >
                  <Grid container>
                    <Grid item className={classes.tabPanelContainer}>
                      <Grid container direction="row" justifyContent="space-between" spacing={1}>
                        <Grid item>
                          <Typography variant="h6" component="h1">
                            {title}
                          </Typography>
                          <Typography className={classes.text}>{subtitle}</Typography>
                        </Grid>
                        <Grid id="terminalFlow" item onClick={setFlow}>
                          <Box display="flex" alignItems="center" justifyContent="flex-end">
                            {!flow && <KeyboardArrowLeftIcon className={classes.flowIcon} />}
                            <Button color="primary" type="button" className={classes.flowButton}>
                              {flow ? 'Hide' : 'Show'} flow
                            </Button>
                            {flow && <KeyboardArrowRightIcon className={classes.flowIcon} />}
                          </Box>
                        </Grid>
                      </Grid>
                    </Grid>
                  </Grid>
                  <Grid>{children}</Grid>
                </div>
                <Grid className={receiptToggle ? classes.receiptPanelOpened : classes.receiptPanel}>
                  {receiptToggle && <ReceiptPanel />}
                </Grid>
              </Grid>
            </Container>
          </div>
          <FlowPanel flow={flow} terminal={terminal} />
        </div>
      )}
    </>
  );
}
