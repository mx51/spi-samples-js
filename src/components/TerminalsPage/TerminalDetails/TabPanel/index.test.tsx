import React from 'react';
import { cleanup } from '@testing-library/react';
import TabPanel from '.';
import mockWithRedux, { defaultMockTerminals, mockTerminalInstanceId } from '../../../../utils/tests/common';

describe('Test <TabPanel />', () => {
  const handleDrawerToggle = jest.fn();

  afterEach(cleanup);

  test('should match TabPanel snapshot test', () => {
    // Arrange
    const container = mockWithRedux(
      <TabPanel
        flow={false}
        index={0}
        setFlow={handleDrawerToggle()}
        subtitle="Test tab 1 subtitle"
        title="Test tab 1 title"
        value={0}
        receiptToggle={false}
        terminal={defaultMockTerminals[mockTerminalInstanceId]}
      >
        <div>Test Panel One</div>
      </TabPanel>
    );

    // Assert
    expect(container).toMatchSnapshot();
    expect(container.innerHTML.includes('<div>Test Panel One</div>')).toBeTruthy();
    expect(handleDrawerToggle).toHaveBeenCalled();
  });

  test('should show receipt panel when receiptToggle value is true', () => {
    const container = mockWithRedux(
      <TabPanel
        flow
        index={1}
        setFlow={handleDrawerToggle()}
        subtitle="Test tab 2 subtitle"
        title="Test tab 2 title"
        value={1}
        receiptToggle
        terminal={defaultMockTerminals[mockTerminalInstanceId]}
      >
        <div>Test Panel Two</div>
      </TabPanel>
    );

    expect(container.innerHTML.includes('Receipt')).toBeTruthy();
    expect(container.innerHTML.includes('Test Panel Two')).toBeTruthy();
    expect(handleDrawerToggle).toHaveBeenCalled();
  });
});
