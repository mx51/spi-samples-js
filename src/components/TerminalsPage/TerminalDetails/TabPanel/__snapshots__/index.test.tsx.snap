// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <TabPanel /> should match TabPanel snapshot test 1`] = `
<div>
  <div
    class="makeStyles-root-1 makeStyles-root-14"
  >
    <div
      class="makeStyles-panel-2"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg"
      >
        <div
          class="MuiGrid-root makeStyles-detailsPanelContainer-9"
        >
          <div
            class="makeStyles-detailsPanel-10"
          >
            <div
              class="MuiGrid-root MuiGrid-container"
            >
              <div
                class="MuiGrid-root makeStyles-tabPanelContainer-4 MuiGrid-item"
              >
                <div
                  class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-justify-content-xs-space-between"
                >
                  <div
                    class="MuiGrid-root MuiGrid-item"
                  >
                    <h1
                      class="MuiTypography-root MuiTypography-h6"
                    >
                      Test tab 1 title
                    </h1>
                    <p
                      class="MuiTypography-root makeStyles-text-6 MuiTypography-body1"
                    >
                      Test tab 1 subtitle
                    </p>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item"
                    id="terminalFlow"
                  >
                    <div
                      class="MuiBox-root MuiBox-root-16"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root makeStyles-flowIcon-8"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M15.41 16.59L10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z"
                        />
                      </svg>
                      <button
                        class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-flowButton-7 MuiButton-textPrimary"
                        tabindex="0"
                        type="button"
                      >
                        <span
                          class="MuiButton-label"
                        >
                          Show
                           flow
                        </span>
                        <span
                          class="MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root"
            >
              <div>
                Test Panel One
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root makeStyles-receiptPanel-12"
          />
        </div>
      </div>
    </div>
    <div
      class="MuiDrawer-root MuiDrawer-docked makeStyles-drawer-17"
      data-test-id="flowPanel"
    >
      <div
        class="MuiPaper-root MuiDrawer-paper makeStyles-drawerPaper-18 MuiDrawer-paperAnchorRight MuiDrawer-paperAnchorDockedRight MuiPaper-elevation0"
        style="visibility: hidden; webkit-transform: translateX(1024px) translateX(0px); transform: translateX(1024px) translateX(0px);"
      >
        <div
          class="MuiBox-root MuiBox-root-22 makeStyles-flowBoxWrapper-19"
        >
          <div
            class="MuiBox-root MuiBox-root-23 makeStyles-flowBox-20"
          >
            <h1
              class="MuiTypography-root MuiTypography-h6"
            >
              Flow
            </h1>
            <pre
              class="makeStyles-flowContent-21"
            >
              
              
# ----------- STATUS -----------

# POSID: test
# EFTPOS: 192.168.1.10
# SPI STATUS: Unpaired
# FLOW: null

# ------------------------------
# POS: - Spi: -
  
            </pre>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
